feat(ses): update lockdown init and enable Hermes VM (#2723)
endojs/endo: packages/ses/error-codes/SES_DIRECT_EVAL.md

Author: add error code
`# 'hostEvaluators' was set to 'all', but the Function() constructor and eval() are not allowed to execute (SES_DIRECT_EVAL)`

The SES Hardened JavaScript shim captures the `eval` function when it is
initialized.
The `eval` function it finds must be the original `eval` because SES uses its
dynamic scope to implement its isolated eval.

If you see this error, something running before `ses` initialized, most likely
another instance of `ses`, has replaced `eval` with something else.

If you're running under an environment that doesn't support direct eval (Hermes), try setting `hostEvaluators` to `no-direct`.

If you're running under CSP, try setting `hostEvaluators` to `none`.

# _hostEvaluators_ was set to _none_, but evaluators are not blocked (`SES_DIRECT_EVAL`)

> It seems your CSP allows execution of `eval()`, try setting `hostEvaluators` to `all` or `no-direct`.

Reviewer: ```suggestion
You indicated that you expected evaluators to be blocked (eg. by Content Security Policy) but they actually work. It's either a misunderstanding of `hostEvaluators` option or whatever method was used to block evaluators no longer works.
```

> # _hostEvaluators_ was set to _none_, but evaluators are not blocked (`SES_DIRECT_EVAL`)
> 
> You indicated that you expected evaluators to be blocked (eg. by Content Security Policy) but they actually work. It's either a misunderstanding of `hostEvaluators` option or whatever method was used to block evaluators no longer works.

Author: ```suggestion
# 'hostEvaluators' was set to 'none', but the Function() constructor or eval() are allowed to execute (SES_DIRECT_EVAL)

You indicated that you expected evaluators to be blocked (e.g. by Content Security Policy),
but they actually work. It's either a misunderstanding of `hostEvaluators` option or whatever
method was used to block evaluators no longer works.
```

> # `"hostEvaluators" was set to "no-direct", but direct eval is functional
> 
> If evaluators are working, including direct eval, it seems you've upgraded to a version of host environment that now supports them (future Hermes). Or you're running code intended for Hermes in a different host.

Author: ```suggestion
# 'hostEvaluators' was set to 'no-direct', but direct eval is available (SES_DIRECT_EVAL)

If evaluators are working, including direct eval, it seems you've upgraded to a version of 
the host environment that now supports them (future Hermes). Or you're running code intended
for Hermes in a different host.
```
